pyhoma/enums/ui.py

```diff
@@ -0,0 +1,396 @@
+    STATELESS_EXTERIOR_HEATING = "StatelessExteriorHeating"
+    STATELESS_ON_OFF = "StatelessOnOff"
+    SUN_ENERGY_SENSOR = "SunEnergySensor"
+    SUN_INTENSITY_SENSOR = "SunIntensitySensor"
+    SWIMMING_POOL = "SwimmingPool"
+    SWINGING_SHUTTER = "SwingingShutter"
+    TSKALARM_CONTROLLER = "TSKAlarmController"
+    TEMPERATURE_SENSOR = "TemperatureSensor"
+    THERMOSTAT_HEATING_TEMPERATURE_INTERFACE = "ThermostatHeatingTemperatureInterface"
+    THERMOSTAT_SET_POINT = "ThermostatSetPoint"
+    THERMOSTAT_ZONES_CONTROLLER = "ThermostatZonesController"
+    THREE_WAY_WINDOW_HANDLE = "ThreeWayWindowHandle"
+    TILT_ONLY_VENETIAN_BLIND = "TiltOnlyVenetianBlind"
+    TIMED_ON_OFF = "TimedOnOff"
+    TIMED_ON_OFF_LIGHT = "TimedOnOffLight"
+    UNIVERSAL_SENSOR = "UniversalSensor"
+    UNTYPED = "Untyped"
+    UP_DOWN_BIOCLIMATIC_PERGOLA = "UpDownBioclimaticPergola"
+    UP_DOWN_CELLULAR_SCREEN = "UpDownCellularScreen"
+    UP_DOWN_CURTAIN = "UpDownCurtain"
+    UP_DOWN_DUAL_CURTAIN = "UpDownDualCurtain"
+    UP_DOWN_EXTERIOR_SCREEN = "UpDownExteriorScreen"
+    UP_DOWN_EXTERIOR_VENETIAN_BLIND = "UpDownExteriorVenetianBlind"
+    UP_DOWN_GARAGE_DOOR = "UpDownGarageDoor"
+    UP_DOWN_GARAGE_DOOR_4T = "UpDownGarageDoor4T"
+    UP_DOWN_GARAGE_DOOR_WITH_VENTILATION_POSITION = (
+        "UpDownGarageDoorWithVentilationPosition"
+    )
+    UP_DOWN_HORIZONTAL_AWNING = "UpDownHorizontalAwning"
+    UP_DOWN_ROLLER_SHUTTER = "UpDownRollerShutter"
+    UP_DOWN_SCREEN = "UpDownScreen"
+    UP_DOWN_SHEER_SCREEN = "UpDownSheerScreen"
+    UP_DOWN_SWINGING_SHUTTER = "UpDownSwingingShutter"
+    UP_DOWN_VENETIAN_BLIND = "UpDownVenetianBlind"
+    UP_DOWN_WINDOW = "UpDownWindow"
+    UP_DOWN_ZEBRA_SCREEN = "UpDownZebraScreen"
+    VOCSENSOR = "VOCSensor"
+    VALVE_HEATING_TEMPERATURE_INTERFACE = "ValveHeatingTemperatureInterface"
+    VENTILATION_INLET = "VentilationInlet"
+    VENTILATION_OUTLET = "VentilationOutlet"
+    VENTILATION_TRANSFER = "VentilationTransfer"
+    WATER_DETECTION_SENSOR = "WaterDetectionSensor"
+    WEATHER_FORECAST_SENSOR = "WeatherForecastSensor"
+    WIFI = "Wifi"
+    WIND_SPEED_SENSOR = "WindSpeedSensor"
+    WINDOW_LOCK = "WindowLock"
+    WINDOW_WITH_TILT_SENSOR = "WindowWithTiltSensor"
+    ZWAVE_AEOTEC_CONFIGURATION = "ZWaveAeotecConfiguration"
+    ZWAVE_CONFIGURATION = "ZWaveConfiguration"
+    ZWAVE_DANFOSS_RSLINK = "ZWaveDanfossRSLink"
+    ZWAVE_DOOR_LOCK_CONFIGURATION = "ZWaveDoorLockConfiguration"
+    ZWAVE_FIBARO_ROLLER_SHUTTER_CONFIGURATION = "ZWaveFibaroRollerShutterConfiguration"
+    ZWAVE_HEATIT_THERMOSTAT_CONFIGURATION = "ZWaveHeatitThermostatConfiguration"
+    ZWAVE_NODON_CONFIGURATION = "ZWaveNodonConfiguration"
+    ZWAVE_QUBINO_CONFIGURATION = "ZWaveQubinoConfiguration"
+    ZWAVE_SEDEVICE_CONFIGURATION = "ZWaveSEDeviceConfiguration"
+    ZWAVE_TRANSCEIVER = "ZWaveTransceiver"
+    ZIGBEE_NETWORK = "ZigbeeNetwork"
+    ZIGBEE_STACK = "ZigbeeStack"
```

Comment:
That’s insane. How can they maintain all these widgets?

Reply:
I don't know haha, but they were already added to the enums. Didn't see that before :o.